src/KnowbaseItem.php

```diff
@@ -126,6 +126,14 @@ public function canViewItem(): bool
             return true;
         }
 
+        if (
+            $this->fields["allow_access_using_token"]
+            && isset($_GET['token'])
+            && hash_equals($this->fields["token"], $_GET['token'])
```

Comment:
I would prefer to not have to rely on `$_GET` here. I do not have time right now to check if we can change this, but this is clearly I want to check before validating this PR.